### PR TITLE
Add filter for classes and methods with annotation `Generated`

### LINKED ISSUE #696: Please exclude all methods annotated with *.Generated (package insensitive) from coverage report
We have LombokGeneratedFilter and GroovyGeneratedFilter now. These filters exclude by certain annotations.

Please consider of filtering method if it's annotated by *.Generated annotation (package insensitive).

### PR DESCRIPTION
**Pull request by Godin**
Fixes #696

### CODE REVIEW

File: org.jacoco.doc/docroot/doc/changes.html [hunk 1]
```diff
@@ -40,6 +40,10 @@ <h3>New Features</h3>
       <code>when</code> expressions that list all cases of <code>sealed class</code>
       is filtered out during generation of report
       (GitHub <a href="https://github.com/jacoco/jacoco/issues/721">#721</a>).</li>
+  <li>Classes and methods annotated with runtime visiable and invisible annotation
```
Comment (marchof) [inline org.jacoco.doc/docroot/doc/changes.html]: Typo: "visiable" -> "visible"

File: org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/AnnotationGeneratedFilter.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+		if (presentIn(methodNode.invisibleAnnotations)
+				|| presentIn(methodNode.visibleAnnotations)) {
+			output.ignore(methodNode.instructions.getFirst(),
+					methodNode.instructions.getLast());
+		}
+
+	}
+
+	private static boolean matches(final String annotation) {
+		return "Generated;".equals(
```
Comment (marchof) [inline org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/AnnotationGeneratedFilter.java]: - Add unit test for inner class case?
- Use precompiled Regex pattern?

Comment (Godin) [reply org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/AnnotationGeneratedFilter.java]: > Use precompiled Regex pattern?

There is no regex :wink: :laughing:

Comment (Godin) [reply org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/AnnotationGeneratedFilter.java]: And absence of regex is on purpose - they are slow even when precompiled.

Comment (Godin) [reply org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/AnnotationGeneratedFilter.java]: > Add unit test for inner class case?

Done

File: org.jacoco.doc/docroot/doc/changes.html (hunk 1 above)
Comment (Godin) [reply org.jacoco.doc/docroot/doc/changes.html]: Fixed

**Pull request merged**